File: python/gemini/p127.py
import io

def http_hello_page(request: io.RawIOBase, response: io.RawIOBase):
    # Read username from the request stream
    username = request.read().decode('utf-8').strip()
    
    # Construct the message
    message = f"Hello {username}"
    
    # Write to the response stream
    response.write(message.encode('utf-8'))

File: javascript/gemini/p127.js
const fs = require('fs');

async function http_hello_page(request, response) {
    let data = '';

    // Read data from the request stream
    for await (const chunk of request) {
        data += chunk;
    }

    const username = data.trim();
    const greeting = `Hello ${username}`;

    // Write the greeting to the response stream
    response.write(greeting);
    response.end();
}